packages/google/src/google-generative-ai-settings.ts

```diff
@@ -52,4 +53,11 @@ Optional. A list of unique safety settings for blocking unsafe content.
       | 'BLOCK_ONLY_HIGH'
       | 'BLOCK_NONE';
   }>;
+
+  /**
+Optional. When enabled, the model will use Google search to ground the response.
+
+@see https://cloud.google.com/vertex-ai/generative-ai/docs/grounding/overview
+   */
+  useSearchGrounding?: boolean;
```

Comment:
is this supported in both genai & vertex? re shadow features, i think we must avoid that. if it's only supported in vertex, can we somehow supply a different interface for vertex? what if a feature is only supported in genai (happens often)

Reply:
Reworked as:

`GoogleGenerativeAISettings` - has the LCD subset common to both
`InternalGoogleGenerativeAISettings` extends above and adds grounding
`GoogleVertexSettings` extends `InternalGoogleGenerativeAISettings`

If there were a GenAI-only feature we can keep the interface names and alter the definitions as needed, either using interface composition differently, or Omit and `&`, etc..

Here's a `v0`-provided example, but until we actually have a GenAI-only feature(s) keeping it the simple inheritance as above seems reasonable.

```
interface BaseSettings {
  // Common settings for both
}

interface GroundingSettings {
  grounding?: GroundingConfig;
}

interface GenAIOnlySettings {
  genAIOnlyFeature: string;
}

type GoogleGenerativeAISettings = BaseSettings & GenAIOnlySettings;
type InternalGoogleGenerativeAISettings = BaseSettings & GenAIOnlySettings & GroundingSettings;
type GoogleVertexSettings = BaseSettings & GroundingSettings & {
  // Vertex-specific settings
};
```